gha-cache/src/api.rs

```diff
@@ -96,6 +100,8 @@ pub struct Api {
     /// The concurrent upload limit.
     concurrency_limit: Arc<Semaphore>,
 
+    circuit_breaker_429_tripped: Arc<AtomicBool>,
```

Comment:
🤔 Why does this need to be `Arc`'d? I would expect the raw `AtomicBool` to be enough....

Reply:
You can't clone an AtomicBool.

Reply:
...and we have to clone it so it can be put into the pool of futures, without holding on to a borrow to the api struct.

Reply:
Right, I forgot that AtomicBool can't be Cloned by itself... I was tricking myself thinking about the "Atomic reference counting" type wrapping an "Atomic bool"